analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/BazelEventsUtil.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 EngFlow Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.engflow.bazel.invocation.analyzer.bazelprofile;
+
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.CAT_GENERAL_INFORMATION;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.CAT_LOCAL_ACTION_EXECUTION;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.CAT_REMOTE_ACTION_CACHE_CHECK;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.CAT_REMOTE_ACTION_EXECUTION;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.CAT_REMOTE_EXECUTION_UPLOAD_TIME;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.CAT_REMOTE_OUTPUT_DOWNLOAD;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.COMPLETE_REMOTE_EXECUTION_UPLOAD_TIME_UPLOAD;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.COMPLETE_REMOTE_EXECUTION_UPLOAD_TIME_UPLOAD_OUTPUTS;
+import static com.engflow.bazel.invocation.analyzer.bazelprofile.BazelProfileConstants.COMPLETE_SUBPROCESS_RUN;
+
+import com.engflow.bazel.invocation.analyzer.traceeventformat.CompleteEvent;
+
+public class BazelEventsUtil {
+  private BazelEventsUtil() {}
+
+  /** The event indicates that an action was executed locally. */
+  public static boolean indicatesLocalExecution(CompleteEvent event) {
+    return CAT_LOCAL_ACTION_EXECUTION.equals(event.category)
+        || CAT_GENERAL_INFORMATION.equals(event.category)
+            && COMPLETE_SUBPROCESS_RUN.equals(event.name);
+  }
+
+  /** The event indicates that an action was executed remotely. */
+  public static boolean indicatesRemoteExecution(CompleteEvent event) {
+    return CAT_REMOTE_ACTION_EXECUTION.equals(event.category);
+  }
+
+  /**
+   * The event documents a remote cache check. This includes cache checks against remote caches
+   * configured by `--remote_cache` or `--disk_cache`.
```

Comment:
nit: I don't know about VSCode, but IntelliJ's JavaDoc renderer doesn't support MarkDown.

```suggestion
   * configured by {@code --remote_cache} or {@code --disk_cache}.
```